src/libmongoc/src/mongoc/mongoc-shared.h

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2021 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MONGOC_SHARED_H
+#define MONGOC_SHARED_H
```

Comment:
`mongoc_shared_ptr` seems like a really useful abstraction, but I vote to make this header private unless there is a use-case to expose it. That gives us the freedom to make backwards breaking changes to struct layouts or function declarations.